pkg/sdk/simpleschema.go

```diff
@@ -0,0 +1,84 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package sdk
+
+import (
+	"fmt"
+	"strings"
+
+	awssdkmodel "github.com/aws-controllers-k8s/code-generator/pkg/api"
+	ackmodel "github.com/aws-controllers-k8s/code-generator/pkg/model"
+	simpleschema "github.com/kro-run/kro/pkg/simpleschema"
+)
+
+// InjectSimpleSchemaShapes processes custom shapes from the top-level custom_shapes
+// section and injects them into the SDK API model.
+// Only string types are supported - all other types will cause a panic.
+func (h *Helper) InjectSimpleSchemaShapes(sdkapi *ackmodel.SDKAPI) error {
+	if len(h.cfg.CustomShapes) == 0 {
+		return nil 
+	}
+
+	for shapeName, fieldsMap := range h.cfg.CustomShapes {
+		// Convert fieldsMap to map[string]interface{} for ToOpenAPISpec
+		schemaObj := make(map[string]interface{})
+		for fieldName, fieldType := range fieldsMap {
+			schemaObj[fieldName] = fieldType
+		}
+
+		openAPISchema, err := simpleschema.ToOpenAPISpec(schemaObj)
+		if err != nil {
+			panic(fmt.Sprintf("Failed to parse schema for shape %s: %v", shapeName, err))
+		}
+
+		shape := &awssdkmodel.Shape{
+			API:           sdkapi.API,
+			ShapeName:     shapeName,
+			Type:          "structure",
+			Documentation: "Generated from SimpleSchema",
+			MemberRefs:    make(map[string]*awssdkmodel.ShapeRef),
+		}
+
+		// Only allow string type fields
+		for fieldName, prop := range openAPISchema.Properties {
+			if prop.Type != "string" && !strings.Contains(prop.Type, "string") {
+				panic(fmt.Sprintf("Field %s in shape %s has non-string type '%s'",
+					fieldName, shapeName, prop.Type))
+			}
+
+			shape.MemberRefs[fieldName] = &awssdkmodel.ShapeRef{
+				API: sdkapi.API,
+				Shape: &awssdkmodel.Shape{
+					API:       sdkapi.API,
+					ShapeName: fmt.Sprintf("%sType", fieldName),
```

Comment:
Curious to understand why we're calling it `%sType`
```suggestion
					ShapeName: fmt.Sprintf("%sType", fieldName),
```